service/src/java/org/apache/hive/service/auth/ldap/LdapSearch.java

```diff
@@ -180,6 +181,8 @@ public List<String> executeCustomQuery(String query) throws NamingException {
    */
   @Override
   public List<String> executeUserAndGroupFilterQuery(String user, String userDn, String groupSearchFilter, String groupBaseDn) throws NamingException {
+    userDn = userDn.replace("\\", "\\5C").replace("*", "\\2A").replace("(", "\\28").replace(")", "\\29").replace("\000", "\\00");
```

Comment:
can you use utils for that like FilterEncoder.encodeFilterValue

Reply:
Good catch! Fixed.